primitives/runtime/src/lib.rs

```diff
@@ -50,13 +50,48 @@ pub type InstanceId = [u8; 4];
 /// The `module_prefix` (arbitrary slice) may be used to generate module-level
 /// "system" account, so you could have separate "system" accounts for currency
 /// exchange, message dispatch and other modules.
-///
-/// The account is not supposed to actually exists on the chain, or to have any funds.
-/// It is only used to
 pub fn bridge_account_id<AccountId>(bridge: InstanceId, module_prefix: &[u8]) -> AccountId
 where
 	AccountId: Decode + Default,
 {
 	let entropy = (module_prefix, bridge).using_encoded(blake2_256);
 	AccountId::decode(&mut &entropy[..]).unwrap_or_default()
 }
+
+/// A trait used to map AccountIds from a source chain to those on a target chain.
+pub trait AccountIdConverter<S, T> {
+	/// Convert the source chain AccountId to a target chain AccountId.
+	///
+	/// A context can be optionally provided in order to provide extra entropy when deriving the
+	/// target account IDs.
+	fn convert(context: Option<&[u8]>, id: S) -> T;
```

Comment:
I wish the signature could just be `fn convert(S) -> T`, but having the `context` is handy (particularly for having that extra context with the root account derivation). Maybe it makes sense to just make this mandatory?

Reply:
TBH I'd rather fallback to previous simple function and just pass an enum indicating whetever we are deriving from an arbitrary account or a root account, something like this:
```
pub enum SourceAccount<T> {
  Root,
  AccountId(T),
}
fn derive_account_id<T>(bridge_id: InstanceId, source_account: SourceAccount<T>) -> H256 {
   match source_account {
      SourceAccount::Root => ("root", bridge_id).using_encoded(blake2_256),
      SourceAccount::AccountId(acc) => ("acc", bridge_id, acc).using_encoded(blake2_256),
   }
}
```
There are multiple reasons why I prefer this:
- `TargetAccountId` conversion should happen externaly (hence we return `H256` here)
- I don't think this should be configurable, it should rather be predictable (see below)
- I expect this function to be re-used (or re-implemented) in UIs or tools, so it should not rely on anything external (like `context`) if it doesn't have to.